akka-cluster-sharding-typed/src/main/resources/reference.conf

```diff
@@ -8,7 +8,9 @@ akka.cluster.sharding {
   number-of-shards = 1000
 }
 
-# FIXME JoinConfigCompatChecker for number-of-shards
+akka.cluster.configuration-compatibility-check.checkers {
+  akka-cluster-sharding-hash-extractor = "akka.cluster.sharding.typed.JoinConfigCompatCheckClusterSharding"
```

Comment:
`akka-cluster-sharding` is taken by `akka.cluster.sharding.JoinConfigCompatCheckSharding`.
The key `akka-cluster-sharding-hash-extractor` points to the fact that
1. The key must be unique
2. This checker's requirement is currently related to the `MessageExtractor` - the 2 hash implementations.
We could instead name the checker class something related to that hash if desired. I don't think it is called for at this time.